### PR TITLE
Pagination

### PR DESCRIPTION
**Pull request by jGundermann**
Add the skip_rows and n_rows capability

### CODE REVIEW

**Comment by jGundermann** [comment]
There is a problem here. Out of bound is not handled. I will look into it.

File: python/fastexcel/__init__.py [hunk 1]
```diff
@@ -67,17 +67,26 @@ def load_sheet_by_name(
...
     ) -> ExcelSheet:
         """Loads a sheet by name.
 
         `header_row` is the index of the row containing the column labels, default index is 0.
         If `None`, the sheet does not have any column labels.
         `column_names` overrides headers found in the document.
         If `column_names` is used `header_row` will be ignored.
+        `n_rows` is the number of rows to parse.
+        `skip_rows` is the line numbers to skip after the headers.
+        If `header_row` is `None` then it skips the number of lines from 0.
```
Comment (lukapeschke) [inline python/fastexcel/__init__.py:81]: Shouldn't `skip_rows` be applied before the headers ? Or is it to support empty lines before and after the header ?

File: src/types/excelsheet.rs [hunk 2]
```diff
@@ -41,11 +41,38 @@ impl Header {
...
+        skip_rows: usize,
+        n_rows: Option<usize>,
+        range: &Range<CalDataType>,
+    ) -> Result<Self> {
+        let max_height = range.height();
+        if max_height < skip_rows {
+            return Err(anyhow!(
+                "To many rows skipped. Max height is {}",
+                max_height
+            ));
```
[the remark this answers is not in the document]
Comment (lukapeschke) [reply src/types/excelsheet.rs:60]: I'd rather have an empty DataFrame in that case.

File: python/fastexcel/__init__.py (hunk 1 above)
Comment (jGundermann) [reply python/fastexcel/__init__.py:81]: I think the API is reasonably easy.
You can use a header row as column labels or not and you can paginate the results.
I did some example in the test : 
```python
sheet = excel_reader.load_sheet(
        0,
        n_rows=1000000,
        skip_rows=1,
        header_row=None,
        column_names=["This", "Is", "Amazing", "Stuff"],
    )
```

Comment (PrettyWood) [reply python/fastexcel/__init__.py:81]: I reckon it's way better after the header. You can already specify the header row.

Comment (lukapeschke) [reply python/fastexcel/__init__.py:81]: Yes, but I feel like we can specify too much information here: both `skip_rows` and `header_row` allow to specify a  number of rows to skip. Maybe we could replace `header` with `list[str] | Literal['read'] | None`. Meaning "That's the column names / read the header from the first row / No header". `skip_rows` would be skipped from the start of the file.

I might be wrong, but I haven't seen a case yet where we need to skip rows after the header.

Pandas implem does not seem to allow skipping rows before and atfer the header:
image

image

Comment (PrettyWood) [reply python/fastexcel/__init__.py:81]: Skip rows after header is useful to have pagination (like a regular offset)

Comment (lukapeschke) [reply python/fastexcel/__init__.py:81]: I'm brain dead this morning, didn't think about pagination at all :man_facepalming:  Forget what I said!

**Comment by jGundermann** [comment]
Made the change so that if skip_rows  > height then it should give back a empty Dataframe. But now that's in place I'm not convinced that this is the best design.
Wouldn't it be more explicit to raise a error so that the user know that he has reached the limit ?

File: python/fastexcel/__init__.py (hunk 1 above)
Comment (jGundermann) [reply python/fastexcel/__init__.py:81]: We have the same behavior in fastexecel. The difference is that we need to be explicit by passing the header_row to `None`. I find it less confusing than pandas.

**Comment by PrettyWood** [comment]
@jGundermann I tend to prefer the error as well. At least it's explicit and people can always catch it to fallback on an empty dataframe

**Comment by lukapeschke** [comment]
I'm fine with both, although this will be a `RuntimeError` for now.

**Comment by jGundermann** [comment]
We should ad maybe a custom errors in the futures. I just saw that you already add it. #45

**Review by PrettyWood — APPROVED** [review]
👏

**Review by lukapeschke — APPROVED** [review]
:medal_sports:

**Review by Sanix-Darker — APPROVED** [review]
🤌🏽 perfecto

**Pull request merged**